backend/wmg/data/query.py

```diff
@@ -127,7 +127,9 @@ def cell_counts(self, criteria: Union[WmgQueryCriteria, FmgQueryCriteria]) -> Da
     # TODO: refactor for readability: https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues
     #  /chanzuckerberg/single-cell-data-portal/2133
     @staticmethod
-    def _query(cube: Array, criteria: Union[WmgQueryCriteria, FmgQueryCriteria], indexed_dims: List[str]) -> DataFrame:
+    def _query(
```

Comment:
if it wouldn't require too much refactoring, it might be nice to add `*` as an argument here, which will require callers to name the arguments that are passed in.

https://stackoverflow.com/questions/2965271/how-can-we-force-naming-of-parameters-when-calling-a-function

Reply:
@joyceyan - I think that is good idea. I presume you mean something like this:

```
def _query(*, cube: Array, criteria: Union[WmgQueryCriteria, FmgQueryCriteria], indexed_dims: List[str]) -> DataFrame
```

which forces all callers to use keyword arguments?

Do you mind if I do this in another commit? Because I think we also need to think about:

- which functions should be called with _keyword-only_ arguments - that is, the _argument names are part of the explicit interface of the function_ and
- which functions should be called by _positional_ arguments (using a `/` to enforce that arguments preceding it should _not be named by the caller and only called positionally_) - that is, the _argument names are NOT part of the explicit interface of the function_

Reply:
yep the code snippet you posted is what i'm imagining. putting it in another PR or commit is fine - that's why i left the comment but still approved the PR 😄 i think any sort of query-related thing we would probably want to enforce keyword only arguments, and i'd also say that generally speaking any function with > 1 argument should probably enforce keyword only arguments (though obviously not a priority to refactor existing code). i can't really think of any reason why positional arguments would be preferred over keyword arguments, but there might be some math-y use cases here that i haven't seen before?